### PR TITLE
fix(atomic): only loads focus-visible polyfill on unsupported browsers

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-1725

https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible#browser_compatibility

### CODE REVIEW

File: packages/atomic/src/global.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    document.body.querySelector(':focus-visible');
+  } catch (error) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    import('focus-visible/dist/focus-visible.min.js');
+  }
+}
+
+export default function () {
+  setCoveoGlobal();
```
Comment (ThibodeauJF) [inline packages/atomic/src/global.ts:19]: Should be an improvement (won't have to load in all interfaces)

File: packages/atomic/src/utils/initialization-utils.tsx [hunk 2]
```diff
@@ -82,10 +82,25 @@ export interface InitializableComponent extends ComponentInterface {
...
+  if (!element.shadowRoot) {
+    return;
+  }
+
+  if (window.applyFocusVisiblePolyfill) {
     window.applyFocusVisiblePolyfill(element.shadowRoot);
+    return;
   }
+
+  window.addEventListener(
```
Comment (ThibodeauJF) [inline packages/atomic/src/utils/initialization-utils.tsx:99]: when lazy loaded this is necessary https://github.com/WICG/focus-visible#lazy-loading

File: packages/atomic/src/global/global.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    import('focus-visible/dist/focus-visible.min.js');
+  }
+}
+
+export default function () {
+  if (!window) {
+    return;
+  }
+
+  setCoveoGlobal();
```
Comment (ThibodeauJF) [inline packages/atomic/src/global/global.ts:49]: Should be an improvement (won't have to load in all interfaces)

**Review by olamothe — COMMENTED** [review]
What would happen if we did * not * load the polyfill at all ?

Crash in old Safari ? Or something else ?
Because if it's the latter and it's not "too bad"... how about we just don't do anything ?

**Comment by ThibodeauJF** [comment]
> Crash in old Safari ? Or something else ?

Old before 15.4 which release 2-3 months ago, but I'm not against removing it in the future

**Pull request merged**